native_client/deepspeech.h

```diff
@@ -4,83 +4,119 @@
 
 #include <cstddef>
 
-typedef struct _DeepSpeechPrivate DeepSpeechPrivate;
+namespace DeepSpeech
+{
 
-class DeepSpeech {
-  private:
-    DeepSpeechPrivate* mPriv;
+  typedef struct _Private Private;
 
-  public:
-    /**
-     * @brief Initialise a DeepSpeech context.
-     *
-     * @param aModelPath The path to the frozen model graph.
-     * @param aNCep The number of cepstrum the model was trained with.
-     * @param aNContext The context window the model was trained with.
-     *
-     * @return A DeepSpeech context.
-     */
-    DeepSpeech(const char* aModelPath, int aNCep, int aNContext);
-    ~DeepSpeech();
+  class Model {
+    private:
+      Private* mPriv;
 
-    /**
-     * @brief Extract MFCC features from a given audio signal and add context.
-     *
-     * Extracts MFCC features from a given audio signal and adds the appropriate
-     * amount of context to run inference with the given DeepSpeech context.
-     *
-     * @param aBuffer A 16-bit, mono raw audio signal at the appropriate sample
-     *                rate.
-     * @param aBufferSize The sample-length of the audio signal.
-     * @param aSampleRate The sample-rate of the audio signal.
-     * @param[out] aMFCC An array containing features, of shape
-     *                   (@p aNFrames, ncep * ncontext). The user is responsible
-     *                   for freeing the array.
-     * @param[out] aNFrames (optional) The number of frames in @p aMFCC.
-     * @param[out] aFrameLen (optional) The length of each frame
-     *                       (ncep * ncontext) in @p aMFCC.
-     */
-    void getMfccFrames(const short* aBuffer,
-                       unsigned int aBufferSize,
-                       int aSampleRate,
-                       float** aMfcc,
-                       int* aNFrames = NULL,
-                       int* aFrameLen = NULL);
+    public:
+      /**
+       * @brief Initialise a DeepSpeech context.
+       *
+       * @param aModelPath The path to the frozen model graph.
+       * @param aNCep The number of cepstrum the model was trained with.
```

Comment:
It is indeed outside of the scope, but reading that part of the code made me think of that. The more we can embed in the model, the better IMHO.